src/Interpreters/Context.cpp

```diff
@@ -3182,7 +3182,6 @@ void Context::setCluster(const String & cluster_name, const std::shared_ptr<Clus
 
 void Context::initializeSystemLogs()
 {
-    auto lock = getLock();
```

Comment:
Why was this lock removed?

Reply:
The function is called once at server initialization and does not require a lock.
But if we use this lock, it leads to a lock order inversion with the initialization of user-defined functions,
which does not lead to anything bad but was detected by TSan.

Reply:
I've returned it back with a comment - btw, it's surprisingly convoluted.